crates/ra_lsp_server/src/conv.rs

```diff
@@ -301,6 +302,104 @@ impl ConvWith<&FoldConvCtx<'_>> for Fold {
     }
 }
 
+impl ConvWith<&WorldSnapshot> for &'static str {
+    type Output = (SemanticTokenType, Vec<SemanticTokenModifier>);
+
+    fn conv_with(self, _world: &WorldSnapshot) -> (SemanticTokenType, Vec<SemanticTokenModifier>) {
+        match self {
```

Comment:
I would also put a function-local `use SemanticTokenType as Stt;` here to decrease verbosity, but I am not sure that everyone likes this style...

Reply:
Yeah, I am not a fan of local `use *` or renames. 

To make this less verbose, I'd ranter

```rust
let tag = match {
    // many trivial branches;
}

let mods = match {
    // copule of non-trivial branches
    _ => vec![]
}
```

It also makes sense to refactor our internal repr to use tag + mods apporach, but that should be done separately.

Reply:
Or 
```rust
let simple_token_type = match self {
    tags::KEYWORD => Stt::KEYWORD,
    tags::SOMETHING => return (Stt::SOMETHING, vec![MODIFIER])
}
return (simple_token_type, vec![])
```